verify/merkle.go

```diff
@@ -7,9 +7,14 @@ import (
 	"github.com/ethereum/go-ethereum/crypto"
 )
 
-// ProcessInclusionProof processes the Merkle root proof
+// ProcessInclusionProof computes the merkle root hash based on the provided leaf and proof, returning the result.
+// An error is returned if the proof param is malformed.
+//
+// NOTE: this method returning a nil error does NOT indicate that the proof is valid. Rather, it merely indicates that
+// the proof was well-formed. The hash returned by this method must be compared to the claimed root hash, to
+// determine if the proof is valid.
```

Comment:
Thanks for writing docstring!!
Is `ProcessInclusionProof` the most accurate name for this method? I dont have full context on where/how its result (the root) is used, but would it make more sense to rename to `ComputeMerkleRoot` or something?

Reply:
Ok just checked its used in `verifyMerkleProof`. Feels like verify should be in merkle.go, and ProcessInclusionProof should be private but Verify public?

Reply:
Can you also add a comment like in the onchain code that if proof is empty the leafHash will be returned.

Reply:
> Can you also add a comment like in the onchain code that if proof is empty the leafHash will be returned.

Requested clarification added in [26d68ca](https://github.com/Layr-Labs/eigenda-proxy/pull/219/commits/26d68ca1b1c7bfe12c678b5c481b1865fd567bbd)

Reply:
Yep makes sense! Thanks for the change :)